internal/cli/streams/privatelink/create_test.go

```diff
@@ -0,0 +1,149 @@
+// Copyright 2025 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package privatelink
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/cli"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/mocks"
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	atlasv2 "go.mongodb.org/atlas-sdk/v20241113004/admin"
+)
+
+const fileName = "test-privateLink.json"
+
+func TestCreateOpts_Run(t *testing.T) {
+	validPrivateLinkConfigFileContents := `
+		{
+			"provider": "Azure",
+			"region": "US_EAST_2",
+			"serviceEndpointId": "/subscriptions/fd01adff-b37e-4693-8497-83ecf183a145/resourceGroups/test-rg/providers/Microsoft.EventHub/namespaces/test-namespace",
+			"dnsDomain": "test-namespace.servicebus.windows.net"
+		}
+	`
+
+	testCases := []struct {
+		name         string
+		fileContents string
+		wantErr      require.ErrorAssertionFunc
+	}{
+		{
+			name:         "no file passed in",
+			fileContents: "",
+			wantErr:      require.Error,
+		},
+		{
+			name: "file does not contain a provider",
+			fileContents: `
+				{
+					"region": "US_EAST_2",
+					"serviceEndpointId": "/subscriptions/fd01adff-b37e-4693-8497-83ecf183a145/resourceGroups/test-rg/providers/Microsoft.EventHub/namespaces/test-namespace",
+					"dnsDomain": "test-namespace.servicebus.windows.net"
+				}
+			`,
+			wantErr: func(tt require.TestingT, err error, _ ...any) {
+				require.ErrorContains(tt, err, "provider missing")
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			fs := afero.NewMemMapFs()
+
+			if len(tc.fileContents) > 0 {
```

Comment:
as above, for strings is usually preffered to check for empty string (tho the compiler generates the same code)
```suggestion
			if tc.fileContents != "" {
```